nautobot/core/utils/lookup.py

```diff
@@ -276,12 +276,9 @@ def get_table_class_string_from_view_name(view_name):
     Returns:
         table_class_name (String): The name of the model table class or None e.g. LocationTable, CircuitTable
     """
-    app_label, model_name = view_name.split(":")
-    model_name = model_name.split("_")[0]
-    model_name = f"{app_label}.{model_name}"
-    # model name should be in the form of "dcim.location".
-    model = get_model_from_name(model_name)
-    view_class = get_view_for_model(model, "List")
+    url = reverse(view_name)
+    match = resolve(url)
+    view_class = match.func.view_class
```

Comment:
After adding some unit tests, this pattern does not seem to work for UIViewSets based views.

Reply:
In other code we've used this construct:

```
                    # For UIViewSet, view_func.cls gets what we need; for an ObjectListView, view_func.view_class is it.
                    view_or_viewset = getattr(view_func, "cls", getattr(view_func, "view_class", None))
````